### PR TITLE
Cleanups and typing of gmlanutils

### PR DESCRIPTION
**Pull request by polybassa**
Added typing to gmlanutils and cleaned up api documentation.

### CODE REVIEW

**Comment by polybassa** [comment]
Great.. `CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in a future release.
  import cryptography` kills my unit tests. I will inspect that in the next days.

**Comment by polybassa** [comment]
Hopefully fixed it.

**Comment by polybassa** [comment]
rebased

File: scapy/contrib/automotive/gm/gmlanutils.py [hunk 1]
```diff
@@ -33,66 +37,74 @@
...
 def _check_response(resp, verbose):
+    # type: (Packet, Optional[bool]) -> bool
     if resp is None:
         if verbose:
             print("Timeout.")
         return False
     if verbose:
         resp.show()
-    return resp.sprintf("%GMLAN.service%") != "NegativeResponse"
+    return cast(bool, resp.service != 0x7f)  # NegativeResponse
```
Comment (gpotter2) [inline scapy/contrib/automotive/gm/gmlanutils.py:49]: If you rebased against master, this cast shouldn't be necessary anymore

File: scapy/contrib/automotive/gm/gmlanutils.py [hunk 2]
```diff
@@ -327,14 +338,19 @@ def GMLAN_ReadMemoryByAddress(sock, addr, length, timeout=None,
         pkt = GMLAN() / GMLAN_RMBA(memoryAddress=addr, memorySize=length)
         resp = sock.sr1(pkt, timeout=timeout, verbose=0)
         if _check_response(resp, verbose):
-            return resp.dataRecord
+            return cast(bytes, resp.dataRecord)
```
Comment (gpotter2) [inline scapy/contrib/automotive/gm/gmlanutils.py:341]: same

**Comment by polybassa** [comment]
Thanks for your feedback. Applied it.

**Comment by polybassa** [comment]
I rebased, but mypy still doesn't like it.

**Comment by gpotter2** [comment]
Ah it could be possible that you need to add automotive modules to.
https://github.com/secdev/scapy/blob/73a0c3b76278aeac8bfd2337703ce81ab166b66b/.config/mypy/mypy.ini#L11
I'm kinda disapointed that it didn't apply for submodules. I wonder if there's an option for that

**Comment by gpotter2** [comment]
Oh you can probably change those lines to use `**` instead of `*`

**Comment by polybassa** [comment]
Sorry, this doesn't help. BTW why is this option `contribs.*` instead of `contrib.*`?

**Comment by gpotter2** [comment]
That's a mistake.

**Comment by polybassa** [comment]
Somehow I can not make this mypy configuration work. I go back to the casts, for now.

**Comment by gpotter2** [comment]
That's alright. I'll figure it out

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**